SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/EventSearchAssetVoltageDisturbances.tsx

```diff
@@ -42,14 +42,24 @@ const EventSearchAssetVoltageDisturbances: React.FC<SEBrowser.IWidget<any>> = (p
     const [data, setData] = React.useState<IDisturbanceData[]>([]);
 
     React.useEffect(() => {
-        if (props.eventID >= 0) loadDisturbancesData(props.eventID);
+        let handle = getDisturbanceData();
+        handle.done((data) => {
+            setData(data);
+        });
+
```

Comment:
you still need the `return () => { if (handle != null && handle.abort != null) handle.abort(); }` to cancel any outstanding requests before sending new ones